src/OrleansProviders/Streams/Memory/MemoryBatchContainer.cs

```diff
@@ -9,10 +7,10 @@
 
 namespace Orleans.Providers
 {
-    [Serializable]
     internal class MemoryBatchContainer<TSerializer> : IBatchContainer
```

Comment:
BTW, the issue I'm having now is precisely surrounding that context - there's no single public entry point into our system that I can pass to a type during deserialization.

What I'd want is some kind of `IOrleansRuntime` which has an `IGrainFactory`, `SerializationManager`, and probably `IServiceProvider`. I added an `IRuntimeClient` property to `SerializationManager` as an entry point into the system, but it's ugly (and a circular reference)